### PR TITLE
Fix to use a new Cloudflare post validate API

### PR DESCRIPTION
**Pull request by iliajie**
Related: https://github.com/virtualmin/virtualmin-pro/pull/38/files

Ref.: virtualmin/virtualmin-pro@793e5dc#commitcomment-124607039

### CODE REVIEW

File: feature-dns.pl [hunk 1]
```diff
@@ -121,6 +121,10 @@ sub setup_dns
 		}
 	my $cfunc = "dnscloud_".$ctype."_create_domain";
 	my ($ok, $msg, $location, $err2) = &$cfunc($d, $info);
+	my $efunc = "dnscloud_".$ctype."_post_validate";
```
Comment (jcameron) [inline feature-dns.pl:124]: Should this still be called if create_domain failed?

File: feature-dns.pl [hunk 2]
```diff
@@ -815,6 +819,8 @@ sub modify_dns
 	else {
 		my $rfunc = "dnscloud_".$ctype."_rename_domain";
 		my ($ok, $id) = &$rfunc($d, $info);
+		my $efunc = "dnscloud_".$ctype."_post_validate";
```
Comment (jcameron) [inline feature-dns.pl:822]: Should this still be called if rename_domain failed?

File: feature-dns.pl (hunk 1 above)
Comment (iliajie) [reply feature-dns.pl:124]: I don't think so, as we can use it in many other cases to handle other errors, right?

File: feature-dns.pl (hunk 2 above)
Comment (iliajie) [reply feature-dns.pl:822]: > Should this still be called if rename_domain failed?

No, as the error messages and codes are the same. No need to duplicate it.

File: feature-dns.pl (hunk 1 above)
Comment (jcameron) [reply feature-dns.pl:124]: Oh I see ... never mind

**Pull request merged**